lib/board.rb

```diff
@@ -29,4 +29,50 @@ def valid_coordinate?(coordinate)
       true
     end
   end
+
+  def valid_placement?(ship, coordinates)
+      if ship.length == coordinates.length && coordinates.all? { |coordinate| valid_coordinate? (coordinate) }
+        if place_ship?(coordinates) == true && is_occupied?(coordinates) == false
+          true
+        else
+          false
+        end
+      else
+        false
+      end
+    end
+
+  def is_occupied?(coordinates)
+    coordinates.any? { |coordinate| !@cells[coordinate].empty? }
```

Comment:
notes to myself:
.any? returns true if at least one element in the block is true
! bang and does the opposite "Are any of the cells not empty?"